packages/sdk/tests/e2e/dataProtectorCore/processProtectedData.test.ts

```diff
@@ -1,31 +1,61 @@
 import { beforeAll, describe, expect, it } from '@jest/globals';
 import { HDNodeWallet, Wallet } from 'ethers';
+import { IExec } from 'iexec';
 import {
   IExecDataProtectorCore,
   ProtectedDataWithSecretProps,
-  getWeb3Provider,
 } from '../../../src/index.js';
 import {
   MAX_EXPECTED_BLOCKTIME,
   MAX_EXPECTED_WEB2_SERVICES_TIME,
+  deployRandomApp,
+  getTestConfig,
 } from '../../test-utils.js';
 
 describe('dataProtectorCore.processProtectedData()', () => {
   let dataProtectorCore: IExecDataProtectorCore;
   let wallet: HDNodeWallet;
   let protectedData: ProtectedDataWithSecretProps;
+  let appAddress: string;
+  let workerpoolAddress: string;
   beforeAll(async () => {
     wallet = Wallet.createRandom();
     dataProtectorCore = new IExecDataProtectorCore(
-      getWeb3Provider(wallet.privateKey)
+      ...getTestConfig(wallet.privateKey)
     );
+    // create app & workerpool
+    const [ethProvider, options] = getTestConfig(wallet.privateKey);
+    appAddress = await deployRandomApp({
+      ethProvider,
+      teeFramework: 'scone',
+    });
+    const iexec = new IExec({ ethProvider }, options.iexecOptions);
+    await iexec.order
+      .createApporder({ app: appAddress, volume: 1000, tag: ['tee', 'scone'] })
+      .then(iexec.order.signApporder)
+      .then(iexec.order.publishApporder);
+    const { address: workerpool } = await iexec.workerpool.deployWorkerpool({
+      description: 'test pool',
+      owner: await iexec.wallet.getAddress(),
+    });
+    workerpoolAddress = workerpool;
+    await iexec.order
+      .createWorkerpoolorder({
+        workerpool: workerpoolAddress,
+        category: 0,
+        volume: 1000,
+        tag: ['tee', 'scone'],
+      })
+      .then(iexec.order.signWorkerpoolorder)
+      .then(iexec.order.publishWorkerpoolorder);
```

Comment:
may be we can use a Promise.all here 🤷🏻‍♂️

Reply:
the same wallet is used in 2 tx, trying to parallelize will result in nonce collision risk, the second option is to use different wallets.